modules/apps/transfer/keeper/grpc_query_test.go

```diff
@@ -261,3 +262,99 @@ func (suite *KeeperTestSuite) TestEscrowAddress() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestTotalEscrowForDenom() {
+	var (
+		req             *types.QueryTotalEscrowForDenomRequest
+		expEscrowAmount math.Int
+	)
+
+	testCases := []struct {
+		msg      string
+		malleate func()
+		expPass  bool
+	}{
+		{
+			"valid native denom with escrow amount < 2^63",
+			func() {
+				req = &types.QueryTotalEscrowForDenomRequest{
+					Denom: sdk.DefaultBondDenom,
+				}
+
+				expEscrowAmount = math.NewInt(100)
+				suite.chainA.GetSimApp().TransferKeeper.SetTotalEscrowForDenom(suite.chainA.GetContext(), sdk.DefaultBondDenom, expEscrowAmount)
+			},
+			true,
+		},
+		{
+			"valid ibc denom with escrow amount > 2^63",
+			func() {
+				denomTrace := types.DenomTrace{
+					Path:      "transfer/channel-0",
+					BaseDenom: sdk.DefaultBondDenom,
+				}
+
+				suite.chainA.GetSimApp().TransferKeeper.SetDenomTrace(suite.chainA.GetContext(), denomTrace)
+				expEscrowAmount, ok := math.NewIntFromString("100000000000000000000")
+				suite.Require().True(ok)
+				suite.chainA.GetSimApp().TransferKeeper.SetTotalEscrowForDenom(suite.chainA.GetContext(), sdk.DefaultBondDenom, expEscrowAmount)
+
+				req = &types.QueryTotalEscrowForDenomRequest{
+					Denom: denomTrace.IBCDenom(),
+				}
+			},
+			true,
+		},
+		{
+			"valid ibc denom treated as native denom",
+			func() {
+				denomTrace := types.DenomTrace{
+					Path:      "transfer/channel-0",
+					BaseDenom: sdk.DefaultBondDenom,
+				}
+
+				req = &types.QueryTotalEscrowForDenomRequest{
+					Denom: denomTrace.IBCDenom(),
+				}
+			},
+			true, // denom trace is not found, so the denom is considered a native token
+		},
+		{
+			"invalid ibc denom treated as valid native denom",
+			func() {
+				req = &types.QueryTotalEscrowForDenomRequest{
+					Denom: "ibc/123",
+				}
+			},
+			true, // the denom is considered a native token
```

Comment:
should we indicate why it is considered a native token? ie because the denom trace isn't set?